client-proxy/services/lineage/service.go

```diff
@@ -0,0 +1,37 @@
+package lineage
+
+import (
+	"github.com/OpenLineage/OpenLineage/client-proxy/database"
+	"github.com/OpenLineage/OpenLineage/client-proxy/logger"
+	"github.com/OpenLineage/OpenLineage/client-proxy/validator"
+	"net/http"
+)
+
+type Service struct {
+	validator          validator.IEventValidator
+	failedEventHandler validator.IFailedEventHandler
+	eventLogger        logger.IEventLogger
+}
+
+type ILineageService interface {
+	CreateLineage(lineageEvent string) int
+}
+
+func (service *Service) CreateLineage(lineageEvent string) int {
```

Comment:
_Minor:_ Our clients and integrations use `emit()` as the method for handling OL events and would recommend to keep things aligned.

Reply:
Well, this is not a client, but a service that holds the business logic of the proxy. `Create` sounds a little misleading but event emission is not the only job it does. Maybe `ProcessLineage` is better? I don't know.

Reply:
As a second thought, the responsibility of this is adding a lineage event to the local database so that the consumer can take it and process it. I think `Create` makes sense.

Reply:
I'd favor `AddLineage(string)` as the event itself isn't being created but rather validated, then added to the underlying datastore (at least that's my feedback, but you can decide on what's best here!)

Reply:
`AddLineage` sounds good. This PR is open too long and I saw you already approved, so I may include the change to the follow up PR.